### PR TITLE
Fix Sorting in Student Exams

### LINKED ISSUE #2886: Sorting in student exams view not working as expected
Sorting by student name in the "student exams" view does not work as expected.
Clicking on "Student" will sort by the `user` object, leading to a misordering, as one would expect that the list is sorted by the full names of the students, that are displayed.

My first guess would be that the issue is in [this line](https://github.com/ls1intum/Artemis/blob/develop/src/main/webapp/app/exam/manage/student-exams/student-exams.component.html#L147), since it seems to sort by `user` rather than `user.name`.

This occurs in version 4.10.0.

### PR DESCRIPTION
**Pull request by FuchsDominik**
Fixes #2886 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

In the student exams view, it was not possible to filter for submissionDate correctly anymore.
When clicking the sort button next to the submission date, the dates stayed the same while other column sorted accordingly. Therefore, student exams seem to have different submissionDates as they actually have

### Description

The date has to be manually formatted in the component.ts, since the artemisDate pipe seems to not be able to apply sorting changes. 

### Steps for Testing

1. Log in to Artemis
2. Move into a exam with student exams (with multiple and different submissionDates)
3. Keep in mind which student exam has which submissionDate
3. Try to sort after submissionDate and check that the sorting works (so that each student exam has its correct submissionDate)

### CODE REVIEW

**Review by tiffca — DISMISSED** [review]
Sorting by submission date works

**Review by JohannesWeiss — APPROVED** [review]
Code looks good to me

**Review by airelawaleria — APPROVED** [review]

**Review by EneaGore — DISMISSED** [review]
Seems fine to me.

File: src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts [hunk 1]
```diff
@@ -350,4 +351,8 @@ export class StudentExamsComponent implements OnInit {
     private onError(error: any) {
         this.jhiAlertService.error(error.errorKey);
     }
+
+    formatDate(date: Moment | Date | undefined) {
+        return date ? moment(date).format('YYYY-DD-MM HH:mm:ss') : '';
```
Comment (krusche) [inline src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts:356]: can you still use the `artemisDate` pipe here? We should avoid defining custom formats all the time.

You can simply inject it using `private artemisDatePipe: ArtemisDatePipe,` in the constructor and then invoke something like

`return this.artemisDatePipe.transform(date)`

**Review by krusche — CHANGES_REQUESTED** [review]
see my inline comment

File: src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts (hunk 1 above)
Comment (FuchsDominik) [reply src/main/webapp/app/exam/manage/student-exams/student-exams.component.ts:356]: Would be a very good idea. However after trying locally, a rather weird behaviour occurs when using the artemisDate pipe. With the pipe, submissionDates in the column all get parsed to the same value. 

Here using the pipe:
Screenshot 2021-02-19 at 22 36 12

After switching to my approach again, the correct dates are shown:
Screenshot 2021-02-19 at 22 37 32

**Review by krusche — APPROVED** [review]

**Pull request merged**